src/test/java/com/melon/portfoliomanager/AssetControllerTest.java

```diff
@@ -0,0 +1,107 @@
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(OutputCaptureExtension.class)
+public class AssetControllerTest {
+
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @MockBean
+    private TransactionRepository transactionRepository;
+
+    @MockBean
+    private PortfolioItemRepository portfolioItemRepository;
+
+    private TransactionDto transactionDto;
+
+
+    @Test
+    void buyAssets_NoSuchUser_ShouldReturnNoContent() {
+
+
+        TransactionDto transactionDto = new TransactionDto("vvp", "IBM", 5.5, 223.1);
+
+        webTestClient.post().uri("/buy/asset")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(transactionDto))
+                .exchange()
+                .expectStatus().isBadRequest()
+                .expectBody(String.class).isEqualTo("There is no user with such username. Enter a valid username.");
+    }
+
+
+    @Test
+    void buyAssets_ValidRequest_ShouldReturnNoContent() {
+
+        when(userRepository.findByUsername(any(String.class))).thenReturn((List.of(new User("vvp", "vpavlov@melon.com", "Georgi", "Ivanov"))));
+
+        TransactionDto transactionDto = new TransactionDto("vvp", "IBM", 5.5, 223.1);
+
+
+        webTestClient.post().uri("/buy/asset")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(transactionDto))
+                .exchange()
+                .expectStatus().isNoContent();
+    }
+
+    @Test
+    void buyAssets_InvalidQuantity_ShouldBadRequest() {
+
+        when(userRepository.findByUsername(any(String.class))).thenReturn((List.of(new User("vvp", "vpavlov@melon.com", "Georgi", "Ivanov"))));
+
+        TransactionDto transactionDto = new TransactionDto("vvp", "IBM", -5.5, 223.1);
+
+
+        webTestClient.post().uri("/buy/asset")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(transactionDto))
+                .exchange()
+                .expectStatus().isBadRequest()
+                .expectBody(String.class).isEqualTo("[\"must be greater than or equal to 0.01\"]");
+    }
+
+    @Test
+    void buyAssets_InvalidPrice_ShouldBadRequest() {
+
+        when(userRepository.findByUsername(any(String.class))).thenReturn((List.of(new User("vvp", "vpavlov@melon.com", "Georgi", "Ivanov"))));
+
+        TransactionDto transactionDto = new TransactionDto("vvp", "IBM", 5.5, -223.1);
+
+
+        webTestClient.post().uri("/buy/asset")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(transactionDto))
+                .exchange()
+                .expectStatus().isBadRequest()
+                .expectBody(String.class).isEqualTo("[\"The price must be a positive number.\"]");
+    }
+
+
+}
```

Comment:
It is best that we have a test validating the behaviour of the endpoint when unexpected error occurs. (i.e. we return 500 internal server error and log appropriate message)